admin/preprint_providers/views.py

```diff
@@ -216,14 +231,47 @@ def post(self, request, *args, **kwargs):
             file_json = json.loads(file_str)
             # make sure not to import an exported access token for SHARE
             cleaned_result = {key: value for key, value in file_json['fields'].iteritems() if key not in FIELDS_TO_NOT_IMPORT_EXPORT}
-            return JsonResponse(cleaned_result)
+            preprint_provider = self.create_or_update_provider(cleaned_result)
+            return redirect('preprint_providers:detail', preprint_provider_id=preprint_provider.id)
 
     def parse_file(self, f):
         parsed_file = ''
         for chunk in f.chunks():
             parsed_file += str(chunk)
         return parsed_file
 
+    def get_page_provider(self):
+        page_provider_id = self.kwargs.get('preprint_provider_id', '')
+        if page_provider_id:
+            return PreprintProvider.objects.get(id=page_provider_id)
+
+    def add_subjects(self, provider, subject_data):
+        from osf.management.commands.populate_custom_taxonomies import migrate
```

Comment:
This concerns me a little. Maybe it's a common Django thing, but I feel like pulling from a management command into web app code is the wrong way around. Ideally, the code would exist inside the web app, and the management command would call that code instead.

That being said, I don't think we'll change it for this, but I will almost certainly be making a refactor follow-up ticket.